website/source/docs/platform/k8s/service-sync.html.md

```diff
@@ -0,0 +1,274 @@
+
+The default service port can be overridden using an annotation:
+
+```yaml
+kind: Service
+apiVersion: v1
+metadata:
+  name: my-service
+  annotations:
+    "consul.hashicorp.com/service-port": "http"
+```
+
+The annotation value may a name of a port (recommended) or an exact port value.
+
+### Service Tags
+
+A service registered in Consul from K8S will always have the tag "k8s" added
+to it. Additional tags can be specified with a comma-separated annotation value
+as shown below. This will also automatically include the "k8s" tag which can't
+be disabled.
+
+```yaml
+kind: Service
+apiVersion: v1
+metadata:
+  name: my-service
+  annotations:
+    "consul.hashicorp.com/service-tags": "primary,foo"
+```
+
+### Service Meta
+
+A service registered in Consul from K8S will set the `external-source` key to
+"kubernetes". This can be used by API consumers, the UI, CLI, etc. to filter
+service instances that are set in k8s. The Consul UI (in Consul 1.2.3 and later)
+will read this value to show a Kubernetes icon next to all externally
+registered services.
+
+Additional metadata can be specified using annotations. The "KEY" below can be
+set to any key. This allows setting multiple meta values:
+
+```yaml
+kind: Service
+apiVersion: v1
+metadata:
+  name: my-service
+  annotations:
+    "consul.hashicorp.com/service-meta-KEY": "value"
+```
+
+## Consul to Kubernetes
+
+This syncs Consul services into first-class Kubernetes services.
+Each Consul service is synced to an
+[ExternalName](https://kubernetes.io/docs/concepts/services-networking/service/#externalname)
+service in Kubernetes. The external name is configured to be the Consul
+DNS entry.
+
+This enables external services to be discovered using native Kubernetes
+tooling. This can be used to ease software migration into or out of Kubernetes,
+across platforms, to and from hosted services, and more.
+
+-> **Requires Consul DNS in Kubernetes:** This feature requires that
+[Consul DNS](/docs/platform/k8s/dns.html) is configured within Kubernetes.
+In the future we hope to remove this requirement by syncing the instance
+addresses directly into service endpoints.
+
+### Sync Enable/Disable
+
+All Consul services visible with the given Consul ACL token to the sync process
+will be synced to Kubernetes.
+
+There is no way to change this behavior per service. For the opposite sync
+direction (Kubernetes to Consul), you can use Kubernetes annotations to disable
+a sync per service. This is not currently possible for Consul to Kubernetes
+sync and the ACL token must be used to limit what services are synced.
+
+In the future, we hope to support per-service configuration.
+
+### Service Name
+
+When a Consul service is synced to Kubernetes, the name of the Kubernetes
+service will match exactly within Kubernetes.
+
+A service name prefix can be specified to the sync program. If specified, this
+prefix will be attached to Consul services within Kubernetes. This defaults to
```

Comment:
An alternate wording of the first part of the `Service Name` section:

"When a Consul service is synced to Kubernetes, the name of the resulting Kubernetes service will exactly match the Consul service name. To change this default exact match behavior, it is possible to specify a prefix to be added to Consul service names within Kubernetes by using the `-k8s-service-prefix` flag. This can also be specified in the Helm configuration."